src/Vault.sol

```diff
@@ -233,16 +233,17 @@ contract Vault {
         allowed[trustedProtocol] = false;
     }
 
-    /*///////////////////////////////////////////////////////////////
+    /* ///////////////////////////////////////////////////////////////
                           MARGIN REQUIREMENTS
-    ///////////////////////////////////////////////////////////////*/
+    /////////////////////////////////////////////////////////////// */
 
     /**
      * @notice Can be called by authorised applications to open or increase a margin position.
      * @param baseCurrency The Base-currency in which the margin position is denominated
      * @param amount The amount the position is increased.
      * @return success Boolean indicating if there is sufficient free margin to increase the margin position
      * @dev All values expressed in the base currency of the vault with same number of decimals as the base currency.
+     * @dev Since increasing margin position is financial activity, liquidation threshold update is done here.
      */
     function increaseMarginPosition(address baseCurrency, uint256 amount)
```

Comment:
Function is implemented very inefficient:
- getFreeMargin() will loop over all assets
- getLiquidationThreshold() will again loop over all assets

Reply:
I am creating a task and assigning for the optimization

Reply:
https://www.notion.so/arcadiafinance/Optimize-increaseMarginPosition-8ccfe20878aa41948be626450e17be88